### ISSUE TITLE
Game of Zones Phase 3 Deception/Attacks

### ISSUE THREAD

**Issue by zmanian** [cosmos/ibc#437]
Iris
https://github.com/irisnet/goz/tree/master/phase-3

P2P validator
https://economy.p2p.org/cosmos-game-of-zones-phase-3-a-deceptive-rootchain/

Vitwit
https://medium.com/regen-network/goz-phase-3-invalidate-real-tokens-by-minting-fake-tokens-275bb91b7678

**Comment by ethanfrey** [comment]
VitWit's attack is based on the fact that native tokens and ibc-remote tokens are handled the same by the bank module. There is no differentiator other than a naming convention.

A larger change, but maybe useful, would be to store a "ibc" flag on the token somewhere, which is checked. So it is no longer a string but a struct. This is probably too breaking everywhere.

A simpler change would simply to limit the valid characters in any tokens that are not minted by IBC. Only IBC module can mint tokens that have a `/` and, in fact, all tokens minted by it must contain at least one `/`. And we get:

```go
func IsIBCToken(denom string) bool {
    return strings.Contains(denom, "/")
}
```

It doesn't seem there is much logic to limit valid denoms currently, but I think this could be useful.

**Comment by cwgoes** [comment]
> VitWit's attack is based on the fact that native tokens and ibc-remote tokens are handled the same by the bank module. There is no differentiator other than a naming convention.

That's right, but I still don't understand why they can't send back the amount of tokens originally sent from the Hub - do you know why that was the case?

> A simpler change would simply to limit the valid characters in any tokens that are not minted by IBC. Only IBC module can mint tokens that have a / and, in fact, all tokens minted by it must contain at least one /.

This is probably a prudent idea.

**Comment by cwgoes** [comment]
This has since been fixed; IBC-module-generated tokens have an IBC-specific denomination.

**Comment by AdityaSripal** [comment]
^Note: This is true assuming that chains do not allow any other module/user to mint tokens with the `ibc` prefix

**Comment by gjermundgaraba** [comment]
This issue was mistakenly reopened during some integration work, so closing again now.
